tests/foreman/cli/test_organization.py

```diff
@@ -252,710 +253,304 @@ def test_positive_search_scoped(self):
-        self.assertNotIn(medium['name'], org['installation-media'])
-
-    @run_only_on('sat')
-    @tier2
-    def test_positive_add_template_by_name(self):
-        """Add a provisioning template to organization by its name
-
-        :id: bd46a192-488f-4da0-bf47-1f370ae5f55c
-
-        :expectedresults: Template is added to the org
-
-        :CaseLevel: Integration
-        """
-        for name in valid_data_list():
-            with self.subTest(name):
-                org = make_org()
-                template = make_template({
-                    'content': gen_string('alpha'),
-                    'name': name,
-                })
-                Org.add_config_template({
-                    'config-template': template['name'],
-                    'name': org['name'],
-                })
-                org = Org.info({'name': org['name']})
-                self.assertIn(
-                    u'{0} ({1})'. format(template['name'], template['type']),
-                    org['templates']
-                )
+            'user-id': user['id'],
+        })
+        Org.add_user({
+            'id': org['id'],
+            'user-id': admin_user['id'],
+        })
+        org_info = Org.info({'id': org['id']})
+        self.assertIn(user['login'], org_info['users'],
+                      "Failed to add user by id")
+        self.assertIn(admin_user['login'], org_info['users'],
+                      "Failed to add admin user by id")
+        if not bz_bug_is_open(1395229):
+            Org.remove_user({
+                'id': org['id'],
+                'user-id': user['id'],
+            })
+            Org.remove_user({
+                'id': org['id'],
+                'user-id': admin_user['id'],
+            })
+            org_info = Org.info({'id': org['id']})
+            self.assertNotIn(user['login'], org_info['users'],
+                             "Failed to remove user by id")
+            self.assertNotIn(admin_user['login'], org_info['users'],
+                             "Failed to remove admin user by id")
 
     @tier2
-    def test_positive_add_template_by_id(self):
-        """Add a provisioning template to organization by its ID
+    def test_positive_add_and_remove_hostgroups(self):
+        """add and remove a hostgroup from an organization
 
-        :id: 4dd119bf-e9e1-4c9a-9b6b-b2c1cc7bc015
+        :id: 34e2c7c8-dc20-4709-a5a9-83c0dee9d84d
+
+        :expectedresults: Hostgroups are handled as expected
 
-        :expectedresults: Template is added to the org
+        :bz: 1395229
+
+        :steps:
+            1. add and remove hostgroup by name
+            2. add and remove hostgroup by id
 
         :CaseLevel: Integration
         """
-        conf_templ = make_template()
         org = make_org()
-        Org.add_config_template({
-            'config-template-id': conf_templ['id'],
+        hostgroup_a = make_hostgroup()
+        hostgroup_b = make_hostgroup()
```

Comment:
this will look like a nitpick at first sight, but I would logically divide all the tests to the env. setup (fixture) and the actual test steps.
e.g. the above 3 lines should go to the `setup` part as they are not really object to test.
- and once we agree on this, I would further suggest to avoid using cli factories for the setup and use nailgun instead. This will bring in additional runtime boost. Just realize that every cli command means at least 1 (but usually 2) ssh sessions + some hammer overhead.

- what do you think?

Reply:
Overall I agree with all, though I think we should approach it as a coordinated project across the cli, something similar to airgun project. So, I support the idea, but I wouldn't block this pr on it, as this is mainly to scratch my current itch with org + I don't want to make it bigger than it is

Reply:
I agree with @pondrejk. These things can be refactored later as most of them CLI tests are not using a nailgun, we can create separate issue add tests which need refactoring like this. The PR intention itself big enough to merge interrelated scenarios.

I also even recommend using pytest module fixtures whenever required to use the same as used in airgun tests. But yeah not in this PR :)